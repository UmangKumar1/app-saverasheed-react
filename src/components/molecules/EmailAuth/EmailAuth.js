import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styled, { css } from 'react-emotion'

import muiTheme from '../../../theme'
import Button from '../../atoms/Button'

const { shape, string, func } = PropTypes

const {
  palette: {
    primary: { main: primaryMain, contrastText: primaryContrastText },
    tertiary: { main: tertiaryMain, contrastText },
  },
} = muiTheme

const EmailForm = styled('form')({
  display: 'flex',
  flexDirection: 'column',
})

const EmailAuthRoot = styled('div')({
  display: 'flex',
  flexDirection: 'column',
  color: tertiaryMain,
  marginBottom: 50,
})

const EmailAuthTitle = styled('h1')({
  color: contrastText,
  textAlign: 'center',
  marginBottom: 30,
})

const EmailInput = styled('input')({
  padding: 10,
  fontSize: 16,
  marginBottom: 10,
})

const buttonCx = css({
  backgroundColor: primaryMain,
  justifyContent: 'center',
  color: primaryContrastText,
  fontSize: 24,
  padding: '16px 8px',
})

class EmailAuth extends Component {
  static propTypes = {
    match: shape({
      params: shape({ authType: string }),
    }).isRequired,
    submit: func.isRequired,
    email: string,
  }

  static defaultProps = {
    email: '',
  }

  state = { email: this.props.email, password: '' }

  handleChange = type => e => this.setState({ [type]: e.target.value })

  handleSubmit = e => {
    e.preventDefault()
    const {
      match: {
        params: { authType },
      },
    } = this.props
    const { email, password } = this.state
    this.props.submit('password', authType, email, password)
  }

  render() {
    const {
      match: {
        params: { authType },
      },
    } = this.props

    const { email, password } = this.state

    return (
      <EmailAuthRoot>
        <EmailAuthTitle>
          {authType === 'signin' ? 'Sign In' : 'Sign Up'} with email
        </EmailAuthTitle>
        <EmailForm>
          <EmailInput
            placeholder="email"
            value={email}
            onChange={this.handleChange('email')}
          />
          <EmailInput
            type="password"
            placeholder="password"
            value={password}
            onChange={this.handleChange('password')}
          />
          <Button className={buttonCx} onClick={this.handleSubmit}>
            Begin Quest!
          </Button>
        </EmailForm>
      </EmailAuthRoot>
    )
  }
}

export default EmailAuth
