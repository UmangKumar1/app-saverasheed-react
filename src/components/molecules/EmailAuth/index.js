export { default, default as EmailAuth } from './EmailAuth'
