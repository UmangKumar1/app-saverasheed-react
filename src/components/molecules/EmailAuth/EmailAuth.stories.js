import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'

import EmailAuth from '.'

storiesOf('EmailAuth', module)
  .add('sign in', () => (
    <EmailAuth
      match={{ url: '/main/login', params: { authType: 'login' } }}
      submit={action('click!')}
    />
  ))
  .add('sign up', () => (
    <EmailAuth
      match={{ url: '/main/signup', params: { authType: 'signup' } }}
      submit={action('click!')}
    />
  ))
