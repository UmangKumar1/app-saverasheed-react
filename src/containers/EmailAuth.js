import { connect } from 'react-redux'
import { compose } from 'recompose'

import EmailAuth from '../components/molecules/EmailAuth'
import { authenticateRequest } from '../modules/auth/actions'
import { subscribe } from '../util/reselect'

const mapStateToProps = subscribe({})

const actions = { submit: authenticateRequest }

export default compose(
  connect(
    mapStateToProps,
    actions,
  ),
)(EmailAuth)
