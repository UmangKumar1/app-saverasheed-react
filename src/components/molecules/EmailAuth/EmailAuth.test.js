import React from 'react'
import { shallow } from 'enzyme'

import EmailAuth from '.'

const wrap = (props = {}) => shallow(<EmailAuth {...props} />)

describe('EmailAuth', () => {
  it('renders children when passed in', () => {
    const wrapper = wrap({
      match: { url: '/main/login', params: { authType: 'login' } },
      submit: jest.fn(),
    })
    expect(wrapper).toMatchSnapshot()
  })
})
