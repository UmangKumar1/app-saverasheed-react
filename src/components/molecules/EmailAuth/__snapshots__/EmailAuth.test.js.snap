// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailAuth renders children when passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EmailAuth
    email=""
    match={
      Object {
        "params": Object {
          "authType": "login",
        },
        "url": "/main/login",
      }
    }
    submit={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(h1)>
          Sign Up
           with email
        </Styled(h1)>,
        <Styled(form)>
          <Styled(input)
            onChange={[Function]}
            placeholder="email"
            value=""
          />
          <Styled(input)
            onChange={[Function]}
            placeholder="password"
            type="password"
            value=""
          />
          <Styled(button)
            className="css-1l21n03"
            onClick={[Function]}
          >
            Begin Quest!
          </Styled(button)>
        </Styled(form)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "Sign Up",
            " with email",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Sign Up",
          " with email",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(input)
              onChange={[Function]}
              placeholder="email"
              value=""
            />,
            <Styled(input)
              onChange={[Function]}
              placeholder="password"
              type="password"
              value=""
            />,
            <Styled(button)
              className="css-1l21n03"
              onClick={[Function]}
            >
              Begin Quest!
            </Styled(button)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onChange": [Function],
              "placeholder": "email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onChange": [Function],
              "placeholder": "password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Begin Quest!",
              "className": "css-1l21n03",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Begin Quest!",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(h1)>
            Sign Up
             with email
          </Styled(h1)>,
          <Styled(form)>
            <Styled(input)
              onChange={[Function]}
              placeholder="email"
              value=""
            />
            <Styled(input)
              onChange={[Function]}
              placeholder="password"
              type="password"
              value=""
            />
            <Styled(button)
              className="css-1l21n03"
              onClick={[Function]}
            >
              Begin Quest!
            </Styled(button)>
          </Styled(form)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Sign Up",
              " with email",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Sign Up",
            " with email",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(input)
                onChange={[Function]}
                placeholder="email"
                value=""
              />,
              <Styled(input)
                onChange={[Function]}
                placeholder="password"
                type="password"
                value=""
              />,
              <Styled(button)
                className="css-1l21n03"
                onClick={[Function]}
              >
                Begin Quest!
              </Styled(button)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onChange": [Function],
                "placeholder": "email",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onChange": [Function],
                "placeholder": "password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Begin Quest!",
                "className": "css-1l21n03",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Begin Quest!",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
